src/document.jl

```diff
@@ -91,7 +91,7 @@ function parse_file(filename::String)
 end
 
 function parse_string(s::String)
-    p = ccall(xmlParseMemory, Xptr, (Ptr{Cchar}, Cint), s, sizeof(s) + 1)
+    p = ccall(xmlParseMemory, Xptr, (Ptr{Cchar}, Cint), pointer(s), sizeof(s) + 1)
```

Comment:
I don't think the existing code should be printing a deprecation warning. This is a bit odd:

```julia
julia> a = "abc"
"abc"

julia> Base.unsafe_convert(Ptr{UInt8}, a)
Ptr{UInt8} @0x00007fc73334b4c0

julia> Base.unsafe_convert(Ptr{Cchar}, a)
WARNING: convert{T}(p::Type{Ptr{T}},a::Array) is deprecated, use convert(p,pointer(a)) instead.
 in depwarn at ./deprecated.jl:40
 in convert at deprecated.jl:29
 in unsafe_convert at deprecated.jl:451
 in unsafe_convert at pointer.jl:21
WARNING: convert(::Type{Ptr}, ::Array{UInt8,1}) methods should be converted to be methods of unsafe_convert
 in depwarn at ./deprecated.jl:40
 in unsafe_convert at deprecated.jl:452
 in unsafe_convert at pointer.jl:21
Ptr{Int8} @0x00007fc73334b4c0
```

This also suggests you may be able to work around this by replacing `Ptr{Cchar}` (which is `Ptr{Int8}`) with `Ptr{UInt8}`.

cc @vtjnash